react-app/src/library/services/headset.ts

```diff
@@ -171,57 +193,54 @@ export default class HeadsetService {
   endAllCalls(): Promise<any> {
     const service = this.selectedImplementation;
     if (!service || !service.isConnected) {
-      this.logger.info('Headset: No vendor headset connected [endAllCalls]'); // TODO: Logger
+      this.logger.info('Headset: No vendor headset connected [endAllCalls]');
       return Promise.resolve();
     }
 
     return this.selectedImplementation.endAllCalls();
   }
 
-  handleDeviceAnsweredCall(event: VendorEvent<EventInfo>) {
+  handleDeviceAnsweredCall(event: VendorEvent<EventInfo>): void | undefined {
+    console.log('event answered Call -> ', event);
     if (event.vendor !== this.selectedImplementation) {
       return;
     }
 
-    this.logger.info('Headset: device answered the call'); // TODO: Logger
-    this.$headsetEvents.next(new HeadsetEvent(HeadsetEventName.DEVICE_ANSWERED_CALL, event));
+    this.logger.info('Headset: device answered the call');
+    this._headsetEvents$.next({ event: 'deviceAnsweredCall', payload: { ...event.body }});
   }
 
-  handleDeviceRejectedCall(event: VendorConversationIdEvent) {
+  handleDeviceRejectedCall(event: VendorConversationIdEvent): void | undefined {
     if (event.vendor !== this.selectedImplementation) {
       return;
     }
 
-    this.logger.info('Headset: device rejected the call'); // TODO: Logger
-    this.$headsetEvents.next(
-      new HeadsetEvent(HeadsetEventName.DEVICE_REJECTED_CALL, event.body.conversationId)
-    );
+    this.logger.info('Headset: device rejected the call');
+    this._headsetEvents$.next({ event: 'deviceRejectedCall', payload: { conversationId: event.body.conversationId }});
   }
 
-  handleDeviceEndedCall(event: VendorEvent<EventInfo>) {
-    this.logger.info('Headset: device ended the call'); // TODO: Logger
-    this.$headsetEvents.next(new HeadsetEvent(HeadsetEventName.DEVICE_ENDED_CALL, event));
+  handleDeviceEndedCall(event: VendorEvent<EventInfo>): void {
+    this.logger.info('Headset: device ended the call');
+    this._headsetEvents$.next({ event: 'deviceEndedCall', payload: { ...event.body } });
+    this._headsetEvents$.next({ event: 'loggableEvent', payload: { ...event.body } });
```

Comment:
This was a workaround I put in place because I noticed an issue after starting a call and then ending said call.  Every subsequent calls would end immediately because the Subject that the consuming app uses to determine the state of the call was still set to the value that says the call had ended.  This isn't an issue with Plantronics because they automatically have a followup event called `CallIdle` that effectively "clears" the subject.  Definitely open to different approaches and suggestions though.  I guess I don't necessarily have to use `loggableEvent` here but just an empty object or blank event value or something like that.  Thoughts?

Reply:
yeah, the loggableEvent is weird. Is this because the consumer subscribes to headset events when it makes a call? If that's the case, we should either *stay subscribed* which would make it so the consumer doesn't get a repeat event, *or* not use a behaviorSubject for the headset events. do we event need one?